packages/devtools_app/lib/src/inspector/flutter/inspector_data_models.dart

```diff
@@ -106,11 +123,33 @@ class LayoutProperties {
                     LayoutProperties(child, copyLevel: copyLevel - 1))
                 ?.toList(growable: false);
 
+  LayoutProperties._({
+    @required this.node,
+    @required this.children,
+    @required this.constraints,
+    @required this.description,
+    @required this.flexFactor,
+    @required this.isFlex,
+    @required this.size,
+  });
+  factory LayoutProperties.lerp(
+      LayoutProperties begin, LayoutProperties end, double t) {
+    return LayoutProperties._(
+      node: end.node,
+      children: end.children,
+      constraints: BoxConstraints.lerp(begin.constraints, end.constraints, t),
+      description: end.description,
+      flexFactor: begin.flexFactor + (begin.flexFactor - end.flexFactor) * t,
+      isFlex: begin.isFlex && end.isFlex,
+      size: Size.lerp(begin.size, end.size, t),
+    );
+  }
+
   final RemoteDiagnosticsNode node;
   final List<LayoutProperties> children;
   final BoxConstraints constraints;
   final String description;
-  final int flexFactor;
+  final num flexFactor;
```

Comment:
is this intended change? I believe flexFactor would just be int

Reply:
To animate across different sizes, we want to use double values.

We want an animation from flex 1 -> flex 2 to go through 1.2, 1.3, 1.7, etc, until it gets to flex 2.